### ISSUE TITLE
[12.0][13.0][14.0]  sale_purchase: Missing purchase order lines on cancellation activity

### ISSUE THREAD

**Issue by ypapouin** [odoo/odoo#67508]
Impacted versions:
12.0, 13.0, 14.0

Steps to reproduce:

- Create a sale order with a purchasable service.
- Cancel the newly created purchase orders.

Current behavior:
A new activity is created with the text: `Exception(s) occurred on the purchase order(s):`
But the `Exception(s):` list is empty, no purchase order lines are rendered.

Expected behavior:
The `Exception(s):` list should be filled with cancelled order lines.

Solution:
The render_context uses `purchase_lines` but the xml template browse `purchase_order_lines`. So the xml or the python code needs a fix.

https://github.com/odoo/odoo/blob/4a61e69ad2a18ef70a9af13682f66283e33d815f/addons/sale_purchase/models/purchase_order.py#L33-L36
https://github.com/odoo/odoo/blob/4a61e69ad2a18ef70a9af13682f66283e33d815f/addons/sale_purchase/data/mail_data.xml#L59-L63

**Comment by pedrobaeza** [comment]
Can you please propose the PR fixing it?

**Comment by ypapouin** [comment]
Yes I could but I'm afraid that it will fall in a black hole like all my other PR.
What is the better way, fixing the python code or the xml template (need module update) ?

**Comment by Yenthe666** [comment]
@ypapouin Python code wins over XML code because it doesn't require module updates. Make a PR and ping us, we'll try to help you get it going :)

**Comment by Feyensv** [comment]
Ping me on the PR and I'll merge it ;).
(Or if you prefer, I can fix it myself).

Thanks for reporting the issue already :).

**Comment by Feyensv** [comment]
@ypapouin There is no need to create a PR for each branch, we have a bot managing the forwards-ports of bug fixes from one branch to the others ;).

**Comment by ypapouin** [comment]
Oh right, didn't know that, that's convenient !
